integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+            theSequel.setId(42L);
+            theSequel.setName("test2");
+            theSequel.setOwner(person);
+            entityManager.merge(theSequel);
+        });
+
+        doInJPA(entityManager -> {
+            SetExpression<Long> union = new BlazeJPAQuery<Long>(entityManager, cbf)
+                    .union(select(document.id).from(document).where(document.id.eq(1337L)),
+                            new BlazeJPAQuery<Long>().intersect(
+                                    select(document.id).from(document).where(document.id.eq(41L)),
+                                    new BlazeJPAQuery<Long>().except(
+                                            select(document.id).from(document).where(document.id.eq(42L)),
+                                            select(document.id).from(document).where(document.id.eq(43L)))),
+                            select(document.id).from(document).where(document.id.eq(46L))
+                    );
+
+            QDocument book2 = new QDocument("secondBook");
+
+            List<Document> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .select(book2)
+                    .from(book2).where(book2.id.in(union))
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testCTE() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .with(idHolderCTE, idHolderCTE.id).as(select(document.id).from(document))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testCTEWithBinds() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .with(idHolderCTE, select(new Binds<IdHolderCTE>().bind(idHolderCTE.id, document.id)).from(document))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testCTEWithBinds2() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                .with(idHolderCTE, select(
+                        bind(idHolderCTE.id, document.id)).from(document))
+                .select(idHolderCTE.id).from(idHolderCTE)
+                .fetch();
+
+                assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQL.class })
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet`
Old MySQL version should work fine because we normally inline the CTE. Any reason to disable this here?